packages/next/src/server/next-server.ts

```diff
@@ -1131,6 +1137,8 @@ export default class NextNodeServer extends BaseServer<
         const isMiddlewareRequest = getRequestMeta(req, 'middlewareInvoke')
 
         const reqCallback = () => {
+          if (this.nextConfig.logging === false) return
```

Comment:
if you did an early return then `delete normalizedReq.fetchMetrics` will not get executed.

Reply:
How about other early returns? Is it ok to move assign & delete the `fetchMetrics` value to the top of reqCallback?
```suggestion
          const fetchMetrics = normalizedReq.fetchMetrics || []
          delete normalizedReq.fetchMetrics

          if (this.nextConfig.logging === false) return
```